tools/gitpod/Dockerfile

```diff
@@ -0,0 +1,59 @@
+FROM gitpod/workspace-base as stage1
+
+COPY --chown=gitpod . /tmp/numpy_repo
+
+# versioneer.py needs a git repo to construct the version
+# We use a multistage build to create a shallow copy of the repo
+RUN git clone --depth 1 file:////tmp/numpy_repo /tmp/numpy
```

Comment:
Do you need to clone the repo? It seems that in the runtime stage you are only using the `environment.yaml` file so might be best to remove multi-stage build and copy the yaml file directly with `COPY` especially if building through GitHub actions

Reply:
While building the runtime image, numpy is built to populate the cache used by ccache. In principle we should be able to build without a `git` history, but `setup.py` depends on `versioneer.py`, which requires a git history. 

The multistage stage is used to have a shallow clone, which reduces the image size of the final image. 

I updated the PR with detailed comments to help clarify this point.